### ISSUE TITLE
Handling of nullable for $ref in OpenAPI 3.0

### ISSUE THREAD

**Issue by bluenote10** [acacode/swagger-typescript-api#39]
I have an API that needs to differentiate between properties that are `T | null` and `T | undefined`. It looks like using `nullable` works for basic types but is ignored for `$ref`:

```yaml
schemas:
  TestObject:
    type: object
    properties:
      stringMaybeUndefined:
        type: string
      stringMaybeNullA:
        type: string
        nullable: true
      stringMaybeNullB:
        anyOf:
        - type: string
        nullable: true
      otherObjectMaybeUndefined:
        $ref: "#/components/schemas/OtherObject"
      otherObjectMaybeNullA:
        $ref: "#/components/schemas/OtherObject"
        nullable: true
      otherObjectMaybeNullB:
        anyOf:
        - $ref: "#/components/schemas/OtherObject"
        nullable: true
    required:
    - stringMaybeNullA
    - stringMaybeNullB
    - otherObjectMaybeNullA
    - otherObjectMaybeNullB

  OtherObject:
    type: object
```

Generates:
```ts
export interface TestObject {
  stringMaybeUndefined?: string;
  stringMaybeNullA?: string | null;
  stringMaybeNullB?: string;
  otherObjectMaybeUndefined?: OtherObject;
  otherObjectMaybeNullA?: OtherObject;
  otherObjectMaybeNullB?: OtherObject;
}
```

Note: The OpenAPI 3.1 syntax
```yaml
oneOf:
- $ref: "#/components/schemas/OtherObject"
- type: 'null'
```
works, but I'm forced to use OpenAPI 3.0 because of constraints from other tooling.

Can this be achieved somehow?

Edit, relevant resources:
- https://github.com/OAI/OpenAPI-Specification/issues/1368
- https://github.com/OAI/OpenAPI-Specification/blob/master/proposals/003_Clarify-Nullable.md

If I interpret these (long) discussions correctly, at least the `allOf` wrapping should work?

**Comment by js2me** [comment]
I know about this problem  
nullable works not very well, I'll fix it today, probably I need 1 hour to fix that  

On my local machine tool currently generates this `TestObject`  
![image](https://user-images.githubusercontent.com/16340911/77634481-164f5f00-6f62-11ea-95b4-fd65e8544045.png)

**Comment by bluenote10** [comment]
Looks good, thanks for the super fast maintenance!

**Comment by js2me** [comment]
@bluenote10 currently I have problem with parsing this case  
```
        otherObjectMaybeNullD:
          anyOf:
          - $ref: "#/components/schemas/OtherObject"
          - type: 'null'
```

I guess it should be like
```
otherObjectMaybeNullD: OtherObject | null;
```

**Comment by bluenote10** [comment]
Yes combining `null` in `anyOf` is a bit awkward, because in this case there cannot be multiple matches. I agree that handling it identical to `oneOf` makes the most sense.

**Comment by js2me** [comment]
I think I covered most cases with usage `nullable` option.  
Schema:  
https://github.com/acacode/swagger-typescript-api/blob/796d1cc7f8fc9a554e30a06b0d6512fd72444b3d/tests/schemas/v3.0/nullable-refs.yaml
Generated ouput:  
https://github.com/acacode/swagger-typescript-api/blob/796d1cc7f8fc9a554e30a06b0d6512fd72444b3d/tests/generated/v3.0/nullable-refs.ts

**Comment by js2me** [comment]
Should be fixed in [release 1.6.3](https://github.com/acacode/swagger-typescript-api/pull/40)

**Comment by Akimyou** [comment]
> I think I covered most cases with usage `nullable` option. Schema: https://github.com/acacode/swagger-typescript-api/blob/796d1cc7f8fc9a554e30a06b0d6512fd72444b3d/tests/schemas/v3.0/nullable-refs.yaml Generated ouput: https://github.com/acacode/swagger-typescript-api/blob/796d1cc7f8fc9a554e30a06b0d6512fd72444b3d/tests/generated/v3.0/nullable-refs.ts

can you handle this part?
```
                      "oneOf": [
                        {
                          "$ref": "#/definitions/348",
                          "x-apifox-overrides": {}
                        },
                        {
                          "type": "null"
                        }
                      ],
```
